### PR TITLE
Windows state and (partial) command implemented

### PR DESCRIPTION
**Pull request by nonninz**

### CODE REVIEW

File: lib/auto_api/commands/windows_command.ex [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    if new_state == state do
+      {:state, state}
+    else
+      {:state_changed, new_state}
+    end
+  end
+
+  def execute(%WindowsState{} = _state, <<0x12, _payload::binary>>) do
+    # TODO
+    raise "Not implemented!"
```
Comment (slashmili) [inline lib/auto_api/commands/windows_command.ex]: not sure if dialyzer likes this. `execute/2` callback is defined in `AutoApi.Command` and it always expects `{:ok, ..} | {:state_change, ..}`

File: lib/auto_api/commands/windows_command.ex [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  """
+  @behaviour AutoApi.Command
+
+  alias AutoApi.WindowsState
+  alias AutoApi.WindowsCapability
+
+  @doc """
+  Parses the binary command and makes changes or returns the state
+
+  TODO: examples
```
Comment (slashmili) [inline lib/auto_api/commands/windows_command.ex]: done forget to add tests/examples here.

File: lib/auto_api/commands/windows_command.ex (hunk 1 above)
Comment (nonninz) [reply lib/auto_api/commands/windows_command.ex]: Hmm dialyzer gives a lot of errors all around the code in this branch so I didn't think about it at all :P

But, looking at it it only issues this complain:
```
lib/auto_api/states/windows_state.ex:32:pattern_match
The pattern
false

can never match the type
true
________________________________________________________________________________
```

Which it says for every State file, so I don't really know what the issue is. It points to the `use AutoApi.State` line

File: lib/auto_api/commands/windows_command.ex (hunk 2 above)
Comment (nonninz) [reply lib/auto_api/commands/windows_command.ex]: Ah true sorry!

**Pull request merged**